src/internal_rep.rs

```diff
@@ -1766,6 +1770,27 @@ impl ValidatedCall {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValidatedModule {
+    pub name: CascadeString,
+    type_infos: Vec<TypeInfo>,
+    validated_modules: Vec<ValidatedModule>,
```

Comment:
Rather than having a ValidatedModule own its children, it would be better to introduce a BTreeSet named ModuleMap that owns all the modules, and an individual ValidatedModule should have references to the module children.  Once we start building, we'll be building arbitrary modules in the tree, and the fact that they happen to have had some higher level structure is irrelevant.

Reply:
Basically the same approach as "TypeMap" and "FunctionMap".